sdk/nodejs/resource.ts

```diff
@@ -77,17 +76,17 @@ export abstract class Resource {
      */
     constructor(t: string, name: string, custom: boolean, props: Inputs = {}, opts: ResourceOptions = {}) {
         if (!t) {
-            throw new RunError("Missing resource type argument");
+            throw new ResourceError("Missing resource type argument", opts.parent);
```

Comment:
notes:

1. these should be not be RunErrors. If we fail to generate a resource, we want to give hte user a stack in their app so they know where the problem came from. Otherwise, if you just got "missing resource name", it would be totally unhelpful as to where they'd need to fix things up.
2. We pass the parent resource as "this" resource hasn't yet been instantiated, and thus cannot be the associated resource here.  Note: thsi should have a good behavior for the user.  They will both get a stack telling them where this happened, and it will be associated with their parent (which is the resource that is trying to create this resource).